libs/core/langchain_core/output_parsers/xml.py

```diff
@@ -47,10 +48,15 @@ def parse(self, text: str) -> Dict[str, List[Any]]:
         if (text.startswith("<") or text.startswith("\n<")) and (
             text.endswith(">") or text.endswith(">\n")
         ):
-            root = ET.fromstring(text)
-            return self._root_to_dict(root)
+            try:
+                root = ET.fromstring(text)
+                return self._root_to_dict(root)
+
+            except ET.ParseError:
```

Comment:
is there any valuable information in the ParseError that we should pass through (eg does it give any detail on what specifically is invalid)

Reply:
Yes, we can check the detailed information in the exception message raised by `ET.fromstring(text)`. 

I updated the code to handle both exceptions while passing through the information in the parent exception message.

In case of mismatched brace in issue, it raises the following exception:
```python
OutputParserException('Failed to parse XML from completion <moviesss>\n    <actor>Tom Hanks</actor></movies>. Got: mismatched tag: line 2, column 30')
```

In case of unexpected string, it raises the following exception:
```python
OutputParserException('Failed to parse XML from completion movie actor: Tom Hanks. Got: syntax error: line 1, column 0')
```

For the changed handling structure, refer to the parse part of PydanticOutputParser.

```python
# langchain_core/output_parsers/pydantic.py

    def parse_result(
        self, result: List[Generation], *, partial: bool = False
    ) -> TBaseModel:
        json_object = super().parse_result(result)
        try:
            return self.pydantic_object.parse_obj(json_object)
        except ValidationError as e:
            name = self.pydantic_object.__name__
            msg = f"Failed to parse {name} from completion {json_object}. Got: {e}"
            raise OutputParserException(msg, llm_output=json_object)

```

Reply:
> is there any valuable information in the ParseError that we should pass through (eg does it give any detail on what specifically is invalid)

Yes, we can check the detailed information in the exception message raised by `ET.fromstring(text)`. 

I updated the code to handle both exceptions while passing through the information in the parent exception message.

In case of mismatched brace in issue, it raises the following exception:
```python
OutputParserException('Failed to parse XML format from completion <moviesss>\n    <actor>Tom Hanks</actor></movies>. Got: mismatched tag: line 2, column 30')
```

In case of unexpected string, it raises the following exception:
```python
OutputParserException('Failed to parse XML format from completion movie actor: Tom Hanks. Got: syntax error: line 1, column 0')
```

For the changed handling structure, refer to the parse part of PydanticOutputParser.

```python
# langchain_core/output_parsers/pydantic.py

    def parse_result(
        self, result: List[Generation], *, partial: bool = False
    ) -> TBaseModel:
        json_object = super().parse_result(result)
        try:
            return self.pydantic_object.parse_obj(json_object)
        except ValidationError as e:
            name = self.pydantic_object.__name__
            msg = f"Failed to parse {name} from completion {json_object}. Got: {e}"
            raise OutputParserException(msg, llm_output=json_object)

```